### DISCUSSION TITLE
Problem with Default Render Pipeline

### DISCUSSION THREAD

**Discussion by danmcguiness** [OndrejNepozitek/Edgar-Unity#120] [Q&A] [answered]
Hi,

Firstly, Edgar is awesome!

I'm very new to Unity and I'm in the process of making my first game. I'm having a bit of trouble setting up the Fog of War feature. 

I get this error but I'm using the Default Render Pipeline. 

The Fog of War feature is enabled but the shader was not yet applied. It seems like you are using URP or LWRP and did not add the custom render feature. Please visit the documentation to see which additional steps are needed to enable this feature in URP and LWRP.
UnityEngine.Debug:LogWarning (object)
Edgar.Unity.FogOfWarGrid2D:Update () (at Assets/ThirdParty/Edgar/Runtime/Grid2D/Common/FogOfWar/FogOfWarGrid2D.cs:199)

I am also using Smart 2D lighting so this may have something to do with the problem.
https://assetstore.unity.com/packages/tools/particles-effects/smart-lighting-2d-112535 

Any help would be awesome,
-Dan.

**Comment by OndrejNepozitek** [comment]
Hello!
Can you check if the Fog of War works in the Example scenes? For example in the Gungeon scene or in the FogOfWar scene.

I think you might be missing the "FogOfWarGrid2D.Setup()" call somewhere in your post-processing logic. It needs to be called each time a level is generated. It's described here: https://ondrejnepozitek.github.io/Edgar-Unity/docs/guides/fog-of-war/#add-custom-post-processing-logic

Let me know if it helps!

**Comment by danmcguiness** [comment]
Hi Ondrej,

The Fog Of War works the Gungeon scene and the FogOfWar scene.

I used the scripted Object from Create/Edgar/Examples/Fog of
War/Post-processing. All I did was change the find tag from "Player" to my
characters tag.

I then put the FogOfWarPostProcessing SO file into the Generators Custom
Post Process task.

Am I missing a step somewhere or do I have to change some of the code in
the FogOfWarPostProcessing SO file. I'm very new to scripting so I don't
know what *" missing the "FogOfWarGrid2D.Setup()" call somewhere in your
post-processing logic" *means.

I redid the whole process again and I'm not getting the error message
anymore but the Fog Of War is still not working.

-Dan.

**Reply by OndrejNepozitek** [reply]
So after you redid it there's no warning/error in the console? You have the FogOfWarGrid2D component attached to the main camera, right? And you're using this post-processing script? https://ondrejnepozitek.github.io/Edgar-Unity/docs/guides/fog-of-war/#add-custom-post-processing-logic By "Fog of War is still not working" you mean that nothing is covered in the fog? In the post-processing script you can add some Debug.Log message after `FogOfWarGrid2D.Instance.Setup(generatedLevelRoot, player.transform);` to make sure that it's called after the level is generated. Also, you need to be in the play mode for the fog to work. And really double check any errors in the console.

**Comment by danmcguiness** [comment]
The Fog is there now and has revealed the first room but it doesn't reveal
the next room. I have set up colliders in the corridors with the Fog Of War
Trigger script from the documentation and changed the tag to my players tag
and set it to "Is Trigger"

It doesn't reveal the rooms in play mode.

I'm not sure what to type to do a Debug.Log message, sorry.



On Tue, 25 Apr 2023 at 03:20, Ondřej Nepožitek ***@***.***>
wrote:

> So after you redid it there's no warning/error in the console? You have
> the FogOfWarGrid2D component attached to the main camera, right? And you're
> using this post-processing script?
> https://ondrejnepozitek.github.io/Edgar-Unity/docs/guides/fog-of-war/#add-custom-post-processing-logic
> By "Fog of War is still not working" you mean that nothing is covered in
> the fog? In the post-processing script you can add some Debug.Log message
> after FogOfWarGrid2D.Instance.Setup(generatedLevelRoot, player.transform);
> to make sure that it's called after the level is generated. Also, you need
> to be in the play mode for the fog to work. And really double check any
> errors in the console.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OndrejNepozitek/Edgar-Unity/discussions/120#discussioncomment-5711291>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A46AEXOFB32YX2TAEYLSVG3XC24P5ANCNFSM6AAAAAAXHXKTWA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Dan McGuiness
Writer - Illustrator  - Videographer - Video Editor - Director - Awesome
Dude

Ph: 0401 948 313
Greenlight Comics <http://www.greenlightcomics.com/>
danmcguiness.com.au

IMPORTANT NOTICE
This email, including all attachments, is confidential and copyright. If
you are not the named recipient you must not use or copy or disclose it to
any person.

**Reply by OndrejNepozitek** [reply]
If the rooms are not revealed then it's another kind of "not working". The fog works but your room detection setup probably has some problems because new rooms are not revealed. Did you add FogOfWarExampleTriggerHandler to your corridor room templates? You also need to make sure to add a RigidBody and a Collider to that same game object to which you added FogOfWarExampleTriggerHandler. Also, check if there are any errors in the console when you enter a room. Sometimes, there's a problem with colliders not triggering due to being set up incorrectly. You have to make sure that the collider triggers when the player enters it.

**Comment by danmcguiness** [comment]
Yes, I have done all of that.

I just checked and I had the error messages were turned off in the
console 😅

This is what I get when I walk into the trigger collider. There seems to be
an error but I don't know how to fix it.
[image: image.png]

Here is the collider set up as well if that helps.
[image: image.png]


On Tue, 25 Apr 2023 at 15:38, Ondřej Nepožitek ***@***.***>
wrote:

> If the rooms are not revealed then it's another kind of "not working". The
> fog works but your room detection setup probably has some problems because
> new rooms are not revealed. Did you add FogOfWarExampleTriggerHandler to
> your corridor room templates? You also need to make sure to add a RigidBody
> and a Collider to that same game object to which you added
> FogOfWarExampleTriggerHandler. Also, check if there are any errors in the
> console when you enter a room. Sometimes, there's a problem with colliders
> not triggering due to being set up incorrectly. You have to make sure that
> the collider triggers when the player enters it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OndrejNepozitek/Edgar-Unity/discussions/120#discussioncomment-5715569>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A46AEXL7OXXSIJBA2NW6EULXC5S7HANCNFSM6AAAAAAXHXKTWA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Dan McGuiness
Writer - Illustrator  - Videographer - Video Editor - Director - Awesome
Dude

Ph: 0401 948 313
Greenlight Comics <http://www.greenlightcomics.com/>
danmcguiness.com.au

IMPORTANT NOTICE
This email, including all attachments, is confidential and copyright. If
you are not the named recipient you must not use or copy or disclose it to
any person.

**Reply by OndrejNepozitek** [reply]
It looks like the images don't load for me.

**Comment by danmcguiness** [comment]
oh, the error is...

MissingReferenceException: The object of type 'Transform' has been
destroyed but you are still trying to access it.
Your script should either check if it is null or you should not destroy the
object.
Edgar.Unity.FogOfWarGrid2D.RevealRooms
(System.Collections.Generic.List`1[T] rooms, System.Nullable`1[T]
waveOrigin, System.Boolean revealImmediately) (at
Assets/ThirdParty/Edgar/Runtime/Grid2D/Common/FogOfWar/FogOfWarGrid2D.cs:267)
Edgar.Unity.FogOfWarGrid2D.RevealRoomAndNeighbors
(Edgar.Unity.RoomInstanceGrid2D room, System.Nullable`1[T] waveOrigin,
System.Boolean revealImmediately) (at
Assets/ThirdParty/Edgar/Runtime/Grid2D/Common/FogOfWar/FogOfWarGrid2D.cs:251)
Edgar.Unity.Examples.FogOfWarExample.FogOfWarExampleTriggerHandler.OnTriggerEnter2D
(UnityEngine.Collider2D otherCollider) (at
Assets/ThirdParty/Edgar/Examples/Grid2D/FogOfWarExample/Scripts/FogOfWarExampleTriggerHandler.cs:37)

On Wed, 26 Apr 2023 at 00:24, Ondřej Nepožitek ***@***.***>
wrote:

> It looks like the images don't load for me.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OndrejNepozitek/Edgar-Unity/discussions/120#discussioncomment-5721133>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A46AEXJJVVE2OSS2FGQMPCLXC7QS7ANCNFSM6AAAAAAXHXKTWA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Dan McGuiness
Writer - Illustrator  - Videographer - Video Editor - Director - Awesome
Dude

Ph: 0401 948 313
Greenlight Comics <http://www.greenlightcomics.com/>
danmcguiness.com.au

IMPORTANT NOTICE
This email, including all attachments, is confidential and copyright. If
you are not the named recipient you must not use or copy or disclose it to
any person.

**Reply by OndrejNepozitek** [reply]
In the FogOfWar component on your camera, which `Mode` do you have? `Wave` or `FadeIn`?

**Comment by danmcguiness** [comment]
Its on Fade In.

On Thu, 27 Apr 2023 at 01:11, Ondřej Nepožitek ***@***.***>
wrote:

> In the FogOfWar component on your camera, which Mode do you have? Wave or
> FadeIn?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OndrejNepozitek/Edgar-Unity/discussions/120#discussioncomment-5733381>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A46AEXMHX2IRMJAHPG7XA7LXDE63NANCNFSM6AAAAAAXHXKTWA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Dan McGuiness
Writer - Illustrator  - Videographer - Video Editor - Director - Awesome
Dude

Ph: 0401 948 313
Greenlight Comics <http://www.greenlightcomics.com/>
danmcguiness.com.au

IMPORTANT NOTICE
This email, including all attachments, is confidential and copyright. If
you are not the named recipient you must not use or copy or disclose it to
any person.

**Comment by danmcguiness** [comment]
I tried the Wave mode and got this error


ArgumentException: When the Wave mode is used, either the waveOrigin that
is passed to this method or that is used in the Setup() method must not be
null.
Edgar.Unity.FogOfWarGrid2D.RevealRooms
(System.Collections.Generic.List`1[T] rooms, System.Nullable`1[T]
waveOrigin, System.Boolean revealImmediately) (at
Assets/ThirdParty/Edgar/Runtime/Grid2D/Common/FogOfWar/FogOfWarGrid2D.cs:268)
Edgar.Unity.FogOfWarGrid2D.RevealRoomAndNeighbors
(Edgar.Unity.RoomInstanceGrid2D room, System.Nullable`1[T] waveOrigin,
System.Boolean revealImmediately) (at
Assets/ThirdParty/Edgar/Runtime/Grid2D/Common/FogOfWar/FogOfWarGrid2D.cs:251)
Edgar.Unity.Examples.FogOfWarExample.FogOfWarExampleTriggerHandler.OnTriggerEnter2D
(UnityEngine.Collider2D otherCollider) (at
Assets/ThirdParty/Edgar/Examples/Grid2D/FogOfWarExample/Scripts/FogOfWarExampleTriggerHandler.cs:37)

On Thu, 27 Apr 2023 at 21:00, Dan McGuiness ***@***.***>
wrote:

> Its on Fade In.
>
> On Thu, 27 Apr 2023 at 01:11, Ondřej Nepožitek ***@***.***>
> wrote:
>
>> In the FogOfWar component on your camera, which Mode do you have? Wave
>> or FadeIn?
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/OndrejNepozitek/Edgar-Unity/discussions/120#discussioncomment-5733381>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/A46AEXMHX2IRMJAHPG7XA7LXDE63NANCNFSM6AAAAAAXHXKTWA>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***
>> .com>
>>
>
>
> --
> Dan McGuiness
> Writer - Illustrator  - Videographer - Video Editor - Director - Awesome
> Dude
>
> Ph: 0401 948 313
> Greenlight Comics <http://www.greenlightcomics.com/>
> danmcguiness.com.au
>
> IMPORTANT NOTICE
> This email, including all attachments, is confidential and copyright. If
> you are not the named recipient you must not use or copy or disclose it to
> any person.
>


-- 
Dan McGuiness
Writer - Illustrator  - Videographer - Video Editor - Director - Awesome
Dude

Ph: 0401 948 313
Greenlight Comics <http://www.greenlightcomics.com/>
danmcguiness.com.au

IMPORTANT NOTICE
This email, including all attachments, is confidential and copyright. If
you are not the named recipient you must not use or copy or disclose it to
any person.

**Reply by OndrejNepozitek** [reply]
Yeah, I investigated this yesterday and there was a small bug. If you open the FogOfWarGrid2D.cs file, you can replace the `RevealRooms` method with the following code (it should be around line 260 in that file):
```
        public void RevealRooms(List<RoomInstanceGrid2D> rooms, Vector3? waveOrigin = null, bool revealImmediately = false)
        {
            if (Mode == FogOfWarMode.Wave && waveOrigin == null && WaveOrigin == null)
            {
                throw new ArgumentException($"When the {FogOfWarMode.Wave} mode is used, either the {nameof(waveOrigin)} that is passed to this method or that is used in the Setup() method must not be null.");
            }

            var playerPosition = Mode == FogOfWarMode.Wave
                ? waveOrigin ?? WaveOrigin.transform.position
                : Vector3.zero;
            
            StartCoroutine(RevealRoomCoroutine(rooms, playerPosition, revealImmediately));
        }
```

This will be shipped in the next update. 

If you wanted to use the Wave mode, you need to pass your player transform to the FogOfWar.Setup() function because the Fog needs to know from which direction to reveal the tiles. But the FadeIn mode should be fine if you do that manual patch in the file.

**Comment by danmcguiness** [comment]
It worked!!!

Awesome work!

Thanks heaps for the help.

All the best,
Dan.

On Thu, 27 Apr 2023 at 21:44, Ondřej Nepožitek ***@***.***>
wrote:

> Yeah, I investigated this yesterday and there was a small bug. If you open
> the FogOfWarGrid2D.cs file, you can replace the RevealRooms method with
> the following code (it should be around line 260 in that file):
>
>         public void RevealRooms(List<RoomInstanceGrid2D> rooms, Vector3? waveOrigin = null, bool revealImmediately = false)
>         {
>             if (Mode == FogOfWarMode.Wave && waveOrigin == null && WaveOrigin == null)
>             {
>                 throw new ArgumentException($"When the {FogOfWarMode.Wave} mode is used, either the {nameof(waveOrigin)} that is passed to this method or that is used in the Setup() method must not be null.");
>             }
>
>             var playerPosition = Mode == FogOfWarMode.Wave
>                 ? waveOrigin ?? WaveOrigin.transform.position
>                 : Vector3.zero;
>
>             StartCoroutine(RevealRoomCoroutine(rooms, playerPosition, revealImmediately));
>         }
>
> This will be shipped in the next update.
>
> If you wanted to use the Wave mode, you need to pass your player transform
> to the FogOfWar.Setup() function because the Fog needs to know from which
> direction to reveal the tiles. But the FadeIn mode should be fine if you do
> that manual patch in the file.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OndrejNepozitek/Edgar-Unity/discussions/120#discussioncomment-5742314>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A46AEXJXFAGV3UCQPWMYNCDXDJPIXANCNFSM6AAAAAAXHXKTWA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Dan McGuiness
Writer - Illustrator  - Videographer - Video Editor - Director - Awesome
Dude

Ph: 0401 948 313
Greenlight Comics <http://www.greenlightcomics.com/>
danmcguiness.com.au

IMPORTANT NOTICE
This email, including all attachments, is confidential and copyright. If
you are not the named recipient you must not use or copy or disclose it to
any person.

**Comment by danmcguiness** [comment]
Hi Ondrej,

I have been following the new documentation on how to set up spawning
enemies and locked doors until the enemies are dead.

I got this error when changing the GungeonRoomManager script like it says
to do in the documentation. Is it because I'm missing an assembly
reference? or something else?
[image: image.png]


On Thu, 27 Apr 2023 at 22:11, Dan McGuiness ***@***.***>
wrote:

> It worked!!!
>
> Awesome work!
>
> Thanks heaps for the help.
>
> All the best,
> Dan.
>
> On Thu, 27 Apr 2023 at 21:44, Ondřej Nepožitek ***@***.***>
> wrote:
>
>> Yeah, I investigated this yesterday and there was a small bug. If you
>> open the FogOfWarGrid2D.cs file, you can replace the RevealRooms method
>> with the following code (it should be around line 260 in that file):
>>
>>         public void RevealRooms(List<RoomInstanceGrid2D> rooms, Vector3? waveOrigin = null, bool revealImmediately = false)
>>         {
>>             if (Mode == FogOfWarMode.Wave && waveOrigin == null && WaveOrigin == null)
>>             {
>>                 throw new ArgumentException($"When the {FogOfWarMode.Wave} mode is used, either the {nameof(waveOrigin)} that is passed to this method or that is used in the Setup() method must not be null.");
>>             }
>>
>>             var playerPosition = Mode == FogOfWarMode.Wave
>>                 ? waveOrigin ?? WaveOrigin.transform.position
>>                 : Vector3.zero;
>>
>>             StartCoroutine(RevealRoomCoroutine(rooms, playerPosition, revealImmediately));
>>         }
>>
>> This will be shipped in the next update.
>>
>> If you wanted to use the Wave mode, you need to pass your player
>> transform to the FogOfWar.Setup() function because the Fog needs to know
>> from which direction to reveal the tiles. But the FadeIn mode should be
>> fine if you do that manual patch in the file.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/OndrejNepozitek/Edgar-Unity/discussions/120#discussioncomment-5742314>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/A46AEXJXFAGV3UCQPWMYNCDXDJPIXANCNFSM6AAAAAAXHXKTWA>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***
>> .com>
>>
>
>
> --
> Dan McGuiness
> Writer - Illustrator  - Videographer - Video Editor - Director - Awesome
> Dude
>
> Ph: 0401 948 313
> Greenlight Comics <http://www.greenlightcomics.com/>
> danmcguiness.com.au
>
> IMPORTANT NOTICE
> This email, including all attachments, is confidential and copyright. If
> you are not the named recipient you must not use or copy or disclose it to
> any person.
>


-- 
Dan McGuiness
Writer - Illustrator  - Videographer - Video Editor - Director - Awesome
Dude

Ph: 0401 948 313
Greenlight Comics <http://www.greenlightcomics.com/>
danmcguiness.com.au

IMPORTANT NOTICE
This email, including all attachments, is confidential and copyright. If
you are not the named recipient you must not use or copy or disclose it to
any person.

**Reply by OndrejNepozitek** [reply]
Hey, the image didn't load again so I cannot see the error.

**Comment by danmcguiness** [comment]
oh damn!

I attached it this time as a jpeg.

the errors are:

Assets\ThirdParty\Edgar\Examples\Grid2D\Gungeon\Scripts\GungeonRoomManager.cs(23,21):
error CS0246: The type or namespace name 'GungeonEnemy' could not be found
(are you missing a using directive or an assembly reference?)

Assets\ThirdParty\Edgar\Examples\Grid2D\Gungeon\Scripts\GungeonRoomManager.cs(17,10):
error CS0246: The type or namespace name 'FormerlySerializedAsAttribute'
could not be found (are you missing a using directive or an assembly
reference?)

Assets\ThirdParty\Edgar\Examples\Grid2D\Gungeon\Scripts\GungeonRoomManager.cs(17,10):
error CS0246: The type or namespace name 'FormerlySerializedAs' could not
be found (are you missing a using directive or an assembly reference?)

On Mon, 15 May 2023 at 02:26, Ondřej Nepožitek ***@***.***>
wrote:

> Hey, the image didn't load again so I cannot see the error.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OndrejNepozitek/Edgar-Unity/discussions/120#discussioncomment-5897960>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A46AEXMSHLWTHRI7EUY2IFLXGEFE5ANCNFSM6AAAAAAXHXKTWA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Dan McGuiness
Writer - Illustrator  - Videographer - Video Editor - Director - Awesome
Dude

Ph: 0401 948 313
Greenlight Comics <http://www.greenlightcomics.com/>
danmcguiness.com.au

IMPORTANT NOTICE
This email, including all attachments, is confidential and copyright. If
you are not the named recipient you must not use or copy or disclose it to
any person.

**Reply by OndrejNepozitek** [reply]
`FormerlySerializedAsAttribute` is an attribute from Unity and for GungeonEnemy you should have `using Edgar.Unity.Examples.Gungeon;`. You should usually use an IDE that can suggest these changes if some using statements are missing.

**Comment by danmcguiness** [comment]
Oh cool. I added 'using UnityEngine.Serialization;' and that fixed the
Unity errors.

I then added the  using Edgar.Unity.Examples.Gungeon; but it still get the
Gungeon errors.

Hopefully you can see the image I attached.


On Mon, 15 May 2023 at 15:35, Ondřej Nepožitek ***@***.***>
wrote:

> FormerlySerializedAsAttribute is an attribute from Unity and for
> GungeonEnemy you should have using Edgar.Unity.Examples.Gungeon;. You
> should usually use an IDE that can suggest these changes if some using
> statements are missing.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OndrejNepozitek/Edgar-Unity/discussions/120#discussioncomment-5901329>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A46AEXP5HEMHHK2KIBJBMFDXGHBSVANCNFSM6AAAAAAXHXKTWA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Dan McGuiness
Writer - Illustrator  - Videographer - Video Editor - Director - Awesome
Dude

Ph: 0401 948 313
Greenlight Comics <http://www.greenlightcomics.com/>
danmcguiness.com.au

IMPORTANT NOTICE
This email, including all attachments, is confidential and copyright. If
you are not the named recipient you must not use or copy or disclose it to
any person.

**Reply by OndrejNepozitek** [reply]
Unfortunately, still no image.

Can you check that you have the file at "Examples/Grid2D/Gungeon/Scripts/GungeonEnemy.cs"? In theory, you shouldn't even need that using statement because the two files should be in the same namespace. So maybe there's a problem with that GungeonEnemy file not being there? Or you can also check the content of the file if the class is named correctly.

**Comment by danmcguiness** [comment]
I don't have GungeonEnemy.cs.

That could be the problem.


On Mon, 15 May 2023 at 16:03, Ondřej Nepožitek ***@***.***>
wrote:

> Unfortunately, still no image.
>
> Can you check that you have the file at
> "Examples/Grid2D/Gungeon/Scripts/GungeonEnemy.cs"? In theory, you shouldn't
> even need that using statement because the two files should be in the same
> namespace. So maybe there's a problem with that GungeonEnemy file not being
> there? Or you can also check the content of the file if the class is named
> correctly.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OndrejNepozitek/Edgar-Unity/discussions/120#discussioncomment-5901525>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A46AEXJ4XS4QX544C322MCLXGHE2LANCNFSM6AAAAAAXHXKTWA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Dan McGuiness
Writer - Illustrator  - Videographer - Video Editor - Director - Awesome
Dude

Ph: 0401 948 313
Greenlight Comics <http://www.greenlightcomics.com/>
danmcguiness.com.au

IMPORTANT NOTICE
This email, including all attachments, is confidential and copyright. If
you are not the named recipient you must not use or copy or disclose it to
any person.

**Reply by OndrejNepozitek** [reply, marked as answer]
I saw that you have the asset at a custom location. Is it possible that you somehow lost the file when moving the asset? You can always import Edgar in another project and make sure that you have all the files in your current project.

**Reply by danmcguiness** [reply]
I moved the entire Edgar folder into a new folder when I started the project. I never moved individual files so I don't think it would have got deleted then. 

I'll try the import Edgar in another project idea.

**Comment by danmcguiness** [comment]
![error3](https://github.com/OndrejNepozitek/Edgar-Unity/assets/121373277/111dc447-250c-4ac1-8bb1-684e442ae971)

**Reply by OndrejNepozitek** [reply]
You're missing the namespace declaration in this file. 

```
﻿using System.Collections.Generic;
using System.Linq;
using UnityEngine;
using UnityEngine.Serialization;

namespace Edgar.Unity.Examples.Gungeon
{
    #region codeBlock:2d_gungeon_roomManager

    public class GungeonRoomManager : MonoBehaviour
    {
    
```

**Reply by danmcguiness** [reply]
![error4](https://github.com/OndrejNepozitek/Edgar-Unity/assets/121373277/186ae0f1-0393-4ac7-90cb-78457b26053a)

I still get the errors with that added.

**Comment by danmcguiness** [comment]
I installed Edgar on a new project. I then copied all the Gungeon scripts over to my current project. That fixed up all the problems.

Thank you for going through this with me.

**Comment by danmcguiness** [comment]
I'm trying to set up the doors to lock when I enter a room, then I have to kill all the enemies to open them. I was following the documentation as best as I could but I now get these errors each time I enter a room. 
![error5](https://github.com/OndrejNepozitek/Edgar-Unity/assets/121373277/484c36e2-2b60-48c8-b76e-7babc0bb42f8)

I don't know what to do next. I have added the floor collider to the Gungeon Room Manager in my room templates. Do I need to manually add the door prefabs to the Gungeon Room Manager in each room template?

I also am having trouble working out how to set up a Door prefab. The gungeon door script has no options to have open, closed or locked graphic or animations. 

I have a another door prefab made in the More Mountains Top Down Engine but I don't think it will work with the Gungeon scripts.

Any help would be awesome, thanks.

**Reply by OndrejNepozitek** [reply]
Where does the error happen? In the unmodified version of that file, it looks like the error is on the line `GungeonGameManager.Instance.OnRoomEnter(roomInstance);` which would imply that you don't have the GungeonGameManager component in your scene.

And regarding door states - there are already lines of code in the GungeonRoomManager file that make the door prefab active or inactive. Instead of that, you can change that to apply animation and then hide the door. Or anything that you want to do with the door.

**Reply by danmcguiness** [reply]
I get the error each time I move from a room to a corridor. 

I added the Gungeon Game Manager to my scene. I added the Level Graph to it but I'm not sure what to put in the "Canvas" field.

I still get the error even with the Gungeon Game Manager added. 

I also get these two errors when I run the game. These happen just once at soon as the game starts.

Error #1

NullReferenceException: Object reference not set to an instance of an object
Edgar.Unity.Examples.Gungeon.GungeonPostProcessingTask.MovePlayerToSpawn (Edgar.Unity.DungeonGeneratorLevelGrid2D level) (at Assets/ThirdParty/Edgar/Examples/Grid2D/Gungeon/Scripts/Tasks/GungeonPostProcessingTask.cs:146)
Edgar.Unity.Examples.Gungeon.GungeonPostProcessingTask.Run (Edgar.Unity.DungeonGeneratorLevelGrid2D level) (at Assets/ThirdParty/Edgar/Examples/Grid2D/Gungeon/Scripts/Tasks/GungeonPostProcessingTask.cs:18)
Edgar.Unity.PostProcessingTaskGrid2D+<Process>d__5.MoveNext () (at Assets/ThirdParty/Edgar/Runtime/Grid2D/DungeonGenerator/PipelineTasks/PostProcessingTaskGrid2D.cs:77)
Edgar.Unity.PipelineRunner`1+<GetEnumerator>d__1[TPayload].MoveNext () (at Assets/ThirdParty/Edgar/Runtime/Common/Pipeline/PipelineRunner.cs:56)
Edgar.Unity.PipelineRunner`1[TPayload].Run (System.Collections.Generic.IEnumerable`1[T] pipelineTasks, TPayload payload, System.Boolean runDiagnostics) (at Assets/ThirdParty/Edgar/Runtime/Common/Pipeline/PipelineRunner.cs:24)
Edgar.Unity.LevelGeneratorBase`1[TPayload].Generate () (at Assets/ThirdParty/Edgar/Runtime/Grid2D/Common/LevelGeneratorBase.cs:45)
Edgar.Unity.DungeonGeneratorBaseGrid2D.Start () (at Assets/ThirdParty/Edgar/Runtime/Grid2D/DungeonGenerator/DungeonGeneratorBaseGrid2D.cs:74)


Error #2

NullReferenceException: Object reference not set to an instance of an object
Edgar.Unity.Examples.Gungeon.GungeonCurrentRoomHandler.Start () (at Assets/ThirdParty/Edgar/Examples/Grid2D/Gungeon/Scripts/GungeonCurrentRoomHandler.cs:14)

**Reply by OndrejNepozitek** [reply]
I think the problem is basically that you have your own scene but want to use parts of the Gungeon scene as you see fit. Usually, the recommended approach is to either use the Gungeon scene fully or make your own scene but also make all the parts from scratch as described in various guides in the docs. But now when you have only some cherry-picked parts of the Gungeon scene, you get these errors because some dependencies are missing. So not sure how to fix this as there can always be another error once you fix the current ones. You should also be able to just open the source file where the error happens, go to that line of code and see why it's null. It should usually be quite easy to fix.
